### PR TITLE
Add rust build

### PR DESCRIPTION
**Pull request by aoudiamoncef**
- [x]  Build ABIs
- [x]  Build APIs
- [x]  CI

### CODE REVIEW

**Comment by modship** [comment]
Hi @aoudiamoncef , i have added massa-xtask module in testnet_23. You can add your task into it and run it with your build.rs.
https://github.com/massalabs/massa/pull/4005

File: build.bat [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@echo off
```
Comment (bilboquet) [inline build.bat:1]: I suggest we use xtask (see https://github.com/massalabs/massa/pull/4005)
this way we could have only one version of build.xxx
what do you think ?

File: rust/build.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            .build_client(true)
+            .type_attribute(
+                ".google.api.HttpRule",
+                "#[cfg(not(doctest))]\n\
+                 #[allow(dead_code)]\n\
+                 pub struct HttpRuleComment{}\n\
+                 /// HACK: see docs in [`HttpRuleComment`] ignored in doctest pass",
+            )
+            .file_descriptor_set_path("src/api.bin")
+            .include_file("_includes.rs")
```
Comment (bilboquet) [inline rust/build.rs:43]: see previous comment
maybe `.include_file("_includes.rs")` => `.include_file("lib.rs")`

**Comment by aoudiamoncef** [comment]
> > Hi @aoudiamoncef , i have added massa-xtask module in testnet_23. You can add your task into it and run it with your build.rs. [massalabs/massa#4005](https://github.com/massalabs/massa/pull/4005)
> > Sorry, i see only now that is not massa repository
> 
> we can't use "your" xtask here but xtask would allow to remove build.bat and build.sh so it's a good idea (and xtask is cheap)

This script will be used to rebuild all projects. Rust, AS, Go .. etc

**Review by bilboquet — COMMENTED** [review]
From my point of view it can be merged 

Two options here:
- Merge then rename the project if we choose so
- Wait for a consensus about the name then merge

I, personally, would go for a merge then explore use of xtask while the name is discussed (maybe) but I feel like Damir is trusting us on this point (cf #core «Do what you feel is right on that naming 🙂»)
So 

Up to you (to merge or wait) :)

File: build.bat (hunk 1 above)
Comment (bilboquet) [reply build.bat:1]: Closed in favor of a more recent comment that supersede this one

**Review by modship — APPROVED** [review]

**Pull request merged**